packages/back-end/src/routers/vercel-native-integration/vercel-native-integration.controller.ts

```diff
@@ -524,10 +606,22 @@ export async function getProducts(req: Request, res: Response) {
 
   if (!slug) return res.status(400).send("Invalid request!");
 
-  if (integration.billingPlanId !== FREE_BILLING_PLAN.id)
-    return res.json({ plans: [NON_FREE_BILLING_PLAN] });
+  const existingBillingPlan = billingPlans.find(
+    ({ id }) => id === integration.billingPlanId
+  );
+
+  if (!existingBillingPlan) {
+    return res.json({ plans: billingPlans });
+  }
+
+  const additionalProjectBillingPlan: BillingPlan = {
+    ...existingBillingPlan,
+    name: "Create GrowthBook Project",
```

Comment:
I think that this would be more clear:
```suggestion
    name: "Add GrowthBook Project to your existing plan",
```